compass-js-frontend/src/main/webapp/ui/view/editor/hierarchy.js

```diff
@@ -52,10 +52,32 @@ var HierarchyView = AmpersandView.extend({
 		});
 		// what we actually want to save is the internal tree object.
 		this.tree = $tree.fancytree('getTree');
+		$tree.on('contextmenu', this.showContextMenu.bind(this));
+	},
+	registerContextOnMenu: function() {
+		var $menu = $('.basicContextContainer');
+		$menu.on('contextmenu', function() {
+			basicContext.close();
+			return false;
+		});
+	},
+	showContextMenu: function(event, data){
+		var node = $.ui.fancytree.getNode(event);
+		this.selectCurrentNode(node);
+		var items = [
+			{ type: 'item', title: 'Add Node', icon: 'ion-plus-round', fn: this.newChild.bind(this) },
```

Comment:
What's the correct behaviour here? Create a node under the currently _selected_ SceneNode (if any), or under the Node which corresponds to the _cursor location_? Does this change selection?
Although, if we want to count this as "just the context menu", and flesh out the action later, I'm willing to let it slide. :wink: